agent_plugin_builder/vendor_dirs.py

```diff
@@ -120,82 +109,76 @@ def _run_container_with_plugin_dir(image: str, command: str, plugin_dir: Path) -
     :param plugin_dir: Path to the plugin directory.
     :return: Output of the container.
     """
-
     client = docker.from_env()  # type: ignore [attr-defined]
     volumes = {str(plugin_dir): {"bind": "/plugin", "mode": "rw"}}
-    return client.containers.run(image, command=command, volumes=volumes, remove=True)
 
+    uid = 0
+    gid = 0
+    if image == WINDOWS_PLUGIN_BUILDER_IMAGE:
+        import os
+
+        uid = os.getuid()
+        gid = os.getgid()
 
-def generate_common_vendor_dir(build_dir: Path, source_dirname: str, uid: int, gid: int):
+    return client.containers.run(
+        image, command=command, volumes=volumes, remove=True, user=f"{uid}:{gid}"
+    )
```

Comment:
This seems iffy. By default, you run as root. Is there any reason not to always use the user's UID and GID?

Reply:
In the linux image we always run as root. Running with the user's UID and GID throws error like: `pyenv: command not found.`